### PR TITLE
Add context menus for graph titles

### PR DESCRIPTION
**Pull request by marthacryan**
Adds context menus to the graph, x, and y axis titles:
title editors

### CODE REVIEW

File: tests/streamlit_ui_tests/graph.spec.ts [hunk 1]
```diff
@@ -245,6 +245,37 @@ test.describe('Graph Functionality', () => {
     await openPopupAndEditTitle(mito, '.g-ytitle', 'Y axis Title');
   });
 
+  test('Update Y axis title with context menu', async ({ page }) => {
```
Comment (naterush) [inline tests/streamlit_ui_tests/graph.spec.ts:248]: Can we parametrize this test across all titles that have context menus?

File: mitosheet/src/mito/components/taskpanes/Graph/graphUtils.tsx [hunk 2]
```diff
@@ -584,37 +593,47 @@ export const registerClickEventsForGraphElements = (graphOutput: GraphOutput, se
     xtitle.style.pointerEvents = 'all'
     ytitle.style.pointerEvents = 'all'
 
+    /**
+     * Open context menu when right clicked
+     */
+    gtitle.addEventListener('contextmenu', (e: any) => {
+        e.preventDefault();
+        setSelectedGraphElement(getGraphElementInfoFromHTMLElement(gtitle, 'gtitle', graphOutput, 'context-menu'))
```
Comment (naterush) [inline mitosheet/src/mito/components/taskpanes/Graph/graphUtils.tsx:601]: These string literals are pretty hard to parse at first glance. Let's make them a little more descriptive. What does `none` even mean -- how can you set the selected graph element to none, is that different than setting it undefined?

File: mitosheet/src/mito/components/taskpanes/Graph/graphUtils.tsx [hunk 3]
```diff
@@ -584,37 +593,47 @@ export const registerClickEventsForGraphElements = (graphOutput: GraphOutput, se
...
-        })
+        setSelectedGraphElement(getGraphElementInfoFromHTMLElement(ytitle, 'ytitle', graphOutput, 'none'))
     })
 
     /**
      * Open popup when double clicked
      */
     gtitle.addEventListener('dblclick', () => {
-        setSelectedGraphElement(getGraphElementInfoFromHTMLElement(gtitle, 'gtitle', graphOutput))
+        setSelectedGraphElement(getGraphElementInfoFromHTMLElement(gtitle, 'gtitle', graphOutput, 'editor'))
```
Comment (naterush) [inline mitosheet/src/mito/components/taskpanes/Graph/graphUtils.tsx:629]: maybe `popup-text-editor` or something more descriptive; editor is really vague,.

File: mitosheet/src/mito/components/taskpanes/Graph/graphUtils.tsx [hunk 4]
```diff
@@ -31,6 +31,9 @@ export interface GraphElementType {
         top?: number,
         bottom?: number,
     }
+
+    /** Whether to display the popup */
```
Comment (naterush) [inline mitosheet/src/mito/components/taskpanes/Graph/graphUtils.tsx:35]: This comment isn't really helpful. Can you describe the three cases, what they are, and how a user would get there? Just a short sentence each.

**Review by naterush — COMMENTED** [review]
Overall, code design seems pretty reasonable! I think there could be some improvements to the literals used to make the code a bit more parsable at first glance

**Comment by aarondr77** [comment]
This suffers from the same bugs fixed in `popup-position-bug` where the context menu is pushed too far up if the y axis title is long

Can you double check that this bus is resolved for the context menu also before release?

https://github.com/mito-ds/mito/assets/18709905/4dc345ea-d742-40cf-9705-416f66e9a2b3

File: mitosheet/src/mito/components/taskpanes/Graph/graphUtils.tsx [hunk 5]
```diff
@@ -571,16 +571,18 @@ export const getGraphElementInfoFromHTMLElement = (graphElement: Element, elemen
         const titleDivTop = mitoDivBottom - clientRect.top + 10;
         const graphDivTop = mitoDivBottom - graphOutputTop;
         let left = clientRect.left - mitoDivLeft;
+        let bottom = Math.min(titleDivTop, graphDivTop);
         // Special case for ytitle, we don't want the context menu to be on top of the title,
         // so we move it to the right a bit
         if (display === 'context-menu') {
             left += 25;
+            bottom -= 25;
```
Comment (aarondr77) [inline mitosheet/src/mito/components/taskpanes/Graph/graphUtils.tsx:579]: Would an even simpler (and probably better solution for the user) just be to set the top of this yTitle context menu in the vertical middle of the MitoContainer? It should be really easy to set that, right? 

Then it's always in a place that is reasonable for the user to look.
